### ISSUE TITLE
Possibly false UndefinedDocblockClass in the following snippet

### ISSUE THREAD

**Issue by githoober** [vimeo/psalm#5224]
https://psalm.dev/r/52d0b846cc

**Comment by githoober** [comment]
"Docblock-defined class or interface K does not exist". The provided template tag "@psalm-template K of object" is not being recognized properly.

**Comment by fezfez** [comment]
I got the same error, any news ? :sparkles:

**Comment by weirdan** [comment]
I don't think we can support templates in type aliases. E.g. what should happen if you import such a type with `@psalm-import-type tuple from HashSet`?

So the issue here is Psalm not explaining that to you in a meaningful way.

**Comment by fox91** [comment]
I think I ran into the same error: https://psalm.dev/r/f71fc33269

An explanation/solution would help.
